### ISSUE TITLE
Sticky header causes glitches if RecyclerView's child item has mergin or divider

### ISSUE THREAD

**Issue by pankona** [shuhart/StickyHeader#2]
Hello!

I've tried this library and the sample codes worked as I expected. That's great!

However, I noticed that if the items held by RecyclerView has margin (represented as `android:marginTop/Bottom`) or divider (that is provided by using `DividerItemDecoration`), sticky header causes glitches if the margin/divider part contacts to the header.

I dived into the implementation, I guess following part should be corrected not to return but continue to latter part.

```java
    // Here're methods in StickyHeaderItemDecorator.java

    public void onDrawOver(Canvas c, RecyclerView parent, RecyclerView.State state) {

        // omit former part

        View viewOverlappedByHeader = getChildInContact(parent, currentStickyHolder.itemView.getBottom());
        if (viewOverlappedByHeader == null) {
            // ****** if return here, sticky header disappeared. ******
            return;
        }

        // omit latter part
    }

    // "margin" or "divider" contacts to the sticky header, this method probably return null.
    private View getChildInContact(RecyclerView parent, int contactPoint) {
        View childInContact = null;
        for (int i = 0; i < parent.getChildCount(); i++) {
            View child = parent.getChildAt(i);
            if (child.getBottom() > contactPoint) {
                if (child.getTop() <= contactPoint) {
                    // This child overlaps the contactPoint
                    childInContact = child;
                    break;
                }
            }
        }
        return childInContact;
    }
```

I think this is not expected behavior of this library. How do you think?

If `getChildInContact` returns `null`, I guess `onDrawOver` should not return then but should continue to proceed with a view that is "last contacted".

Just my idea, `onDrawOver` method should be like this:

```java
// prepare a view represents "last contacted"
private View lastViewOverlappedByHeader = null;

public void onDrawOver(Canvas c, RecyclerView parent, RecyclerView.State state) {

    // omit former part

    View viewOverlappedByHeader = getChildInContact(parent,     currentStickyHolder.itemView.getBottom());
    if (viewOverlappedByHeader == null) {
        // continue processing with "last contacted"
        if (lastViewOverlappedByHeader != null) {
            viewOverlappedByHeader = lastViewOverlappedByHeader;
        } else {
            return;
        }
    }

    // omit latter part
}
```

If you can agree, I'll try to create a pull request for this issue. Please let me know.
Thanks!

**Comment by shuhart** [comment]
Hi! Yes, it would be wonderful if you created a pr so that I can reproduce the error without going into details you already dealt with.

**Comment by pankona** [comment]
Okay, I will!

**Comment by shuhart** [comment]
Fixed in 1.0.6
